playwright-e2e/tests/osteosarcoma/os1-to-os2-reconsent-workflow.spec.ts

```diff
@@ -0,0 +1,411 @@
+import { test } from 'fixtures/dsm-fixture';
+import { expect } from '@playwright/test';
+import { Navigation, Study, StudyName } from 'dsm/navigation';
+import Select from 'dss/component/select';
+import ParticipantListPage from 'dsm/pages/participant-list-page';
+import { CustomizeView, CustomizeViewID, DataFilter, Label } from 'dsm/enums';
+import { updateAuth0UserPassword } from 'utils/api-utils';
+import { APP } from 'data/constants';
+import HomePage from 'dss/pages/osteo/home-page';
+import { fillSitePassword } from 'utils/test-utils';
+import { login } from 'authentication/auth-osteo';
+import ResearchConsentFormPage from 'dss/pages/osteo/research-consent-page';
+import * as user from 'data/fake-user.json';
+import ConsentAddendumPage from 'dss/pages/osteo/consent-addendump-page';
+import SurveyAboutOsteoPage from 'dss/pages/osteo/survey-about-osteo-page';
+import { generateUserName } from 'utils/faker-utils';
+import SurveyAboutYou from 'dss/pages/osteo/survey-about-you';
+import DashboardPage, { DashboardActivity } from 'dss/pages/dashboard-page';
+import SurveyDataTab from 'dsm/pages/tablist/survey-data-tab';
+import { ActivityVersion, SurveyName } from 'dsm/component/tabs/enums/survey-data-enum';
+import { FamilyHistory, Section } from 'dss/pages/family-history';
+import { getToday } from 'utils/date-utils';
+
+test.describe.serial(`Reconsent an OS1 participant into OS2`, () => {
+  test.slow();
+  const PARTICIPANT_PASSWORD = process.env.OSTEO_USER_PASSWORD as string;
+  const OSTEO_BASE_URL = process.env.OSTEO_BASE_URL as string;
+
+  let navigation: Navigation;
+  let shortID: string;
+  let participantEmail: string;
+
+  test(`OS1: Re-consent workflow`, async ({ page, request }) => {
+    await test.step('Choose an OS1 participant to re-consent', async () => {
+      navigation = new Navigation(page, request);
+      await new Select(page, { label: 'Select study' }).selectOption(StudyName.OSTEO);
+
+      const participantListPage = await navigation.selectFromStudy<ParticipantListPage>(Study.PARTICIPANT_LIST);
+      await participantListPage.waitForReady();
+      shortID = await findOS1ParticipantWhoHasNotReconsented(participantListPage);
+
+      const participantListTable = participantListPage.participantListTable;
+      const participantPage = await participantListTable.openParticipantPageAt({ position: 0 });
+      await participantPage.waitForReady();
+      participantEmail = await participantPage.getEmail();
+      expect(participantEmail).toBeTruthy();
+      console.log(`Participant email is: ${participantEmail}`);
+      console.log(`Participant password is: ${PARTICIPANT_PASSWORD}\n`);
+    });
+
+    await test.step('Update their passsword so that they can be logged into', async () => {
```

Comment:
Just curious, why is password update required? Or is this just so the test uses a specific password?

Reply:
Password update is required because I'm using PW created OS1 participants - this way, even if I previously manually changed their password for some reason, it'll be straightforward to just login as them with an updated password (to make sure the test doesn't run into login issues).